encord/objects/ontology_object_instance.py

```diff
@@ -465,8 +478,47 @@ def set_for_frames(
             reviews: Should only be set by internal functions.
             is_deleted: Should only be set by internal functions.
         """
-        if self._range_only:
-            if not isinstance(coordinates, AudioCoordinates):
+        if range_html is not None:
+            if self._ontology_object.shape != Shape.TEXT:
+                raise LabelRowError(
+                    "Setting range_html of the object instance is only allowed for objects with the 'text' shape"
```

Comment:
Biggest nit of all time / very debatable.
```suggestion
                    f"Setting range_html of the object instance is only allowed for objects with the {Shape.TEXT} shape"
```
Might allow people to better contextualise where the shape comes from.